src/lib.rs

```diff
@@ -107,6 +107,33 @@ pub async fn function_handler(
             crate::process::cloudwatch_logs(cloudwatch_event_log, coralogix_exporter, config)
                 .await?;
         }
+        CombinedEvent::Sqs(sqs_event) => {
+            debug!("SQS Event: {:?}", sqs_event.records[0]);
+            if let Some(message) = &sqs_event.records[0].body {
+                if config.integration_type != IntegrationType::Sqs  {
+                    let records = serde_json::from_str::<serde_json::Value>(message)?;               
+                    debug!("SQS S3 EVENT Detected");
+                    let bucket = records["Records"][0]["s3"]["bucket"]["name"]
```

Comment:
SQS lambda trigger hash batching enabled by default (at least when created via CF / SAM). That means we should handle all records, not just the zeroth one (like we could reasonably do with other type of events)

Reply:
Added For